github-extension/src/browser/github-quick-open-service.ts

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { injectable, inject } from "inversify";
+import { MessageService } from '@theia/core/lib/common/message-service';
+import { QuickOpenItem, QuickOpenModel, QuickOpenMode } from "@theia/core/lib/browser/quick-open/quick-open-model";
+import { QuickOpenService, QuickOpenOptions } from '@theia/core/lib/browser/quick-open/quick-open-service';
+import { open, OpenerService } from "@theia/core/lib/browser";
+import { propertiesUri, GitHubProperties } from './github-properties';
+import { GithubService } from "../common/github-service";
+import { Repository } from "../common/github-model";
+import { Git } from "@theia/git/lib/common";
+import URI from "@theia/core/lib/common/uri";
+import { WorkspaceService } from "@theia/workspace/lib/browser/workspace-service";
+import { FileSystem, FileStat } from "@theia/filesystem/lib/common/filesystem";
+import { SelectionService } from "@theia/core/lib/common";
+import { StatusBar, StatusBarAlignment } from '@theia/core/lib/browser/status-bar/status-bar';
+
+@injectable()
+export class GitHubQuickOpenService {
```

Comment:
should it provide an interface ?

Reply:
For now there is no need to extract an interface from this class